src/ol/array.js

```diff
@@ -38,13 +38,13 @@ ol.array.binaryFindNearest = function(arr, target) {
 
 
 /**
- * Default compare for array sort, will make sure number sort works okay.
+ * Compare function for array sort that is safe for numbers.
  * @param {*} a The first object to be compared.
  * @param {*} b The second object to be compared.
  * @return {number} A negative number, zero, or a positive number as the first
  *     argument is less than, equal to, or greater than the second.
  */
-ol.array.defaultCompare = function(a, b) {
+ol.array.numberSafeCompareFunction = function(a, b) {
```

Comment:
Not sure I get the motivation to rename this.  This works for strings or numbers.  As long as the name isn't exposed to users, I don't think it is a big deal, but I think `ol.array.defaultCompare` was sensible.

Reply:
I decided to propose a new name in #14311.

Reply:
I like it.